common/src/main/java/org/conscrypt/SSLUtils.java

```diff
@@ -60,6 +62,15 @@
             System.getProperty("org.conscrypt.useEngineSocketByDefault", "false"));
     private static final int MAX_PROTOCOL_LENGTH = 255;
 
+    static final CertificateFactory X509_CERT_FACTORY;
+    static {
+        try {
+            X509_CERT_FACTORY = CertificateFactory.getInstance("X.509");
```

Comment:
I'd either use `getInstance` each time or use our own `CertificateFactory`. We've had bugs where some part of our stack is broken, but we could have avoided (or did) by poking a fixed version into the Security providers list. `static` assignment has usually been the culprit of failures. This could even have unintended consequences if this was initialized before we call `Security.insertProvider` in the app.

In Android the `Security` infrastructure maintains its own cache so this will only shave off a few microseconds.